### ISSUE TITLE
Arena Damage ignoring snapshots

### ISSUE THREAD

**Issue by M2VC** [kangarko/CoreArena#67]
Hey there, enjoying the plugin thus far, but I've been running into some strange issues lately. 

A couple of users on my server have thrown together an arena in the sky, high above the main world spawn and other player structures and creations. In the arena are a couple of facades intended to be blown away revealing/removing cover for players to use, however instead what we're seeing is: Walls blow up, blocks simulate and fly around, land and cause the floor of the arena to simulate and thus fall as well.

This is problematic for two reasons: One, the floor is not defined in the damaged snapshot, so from our understanding it should not be included in any simulation nor should it take any damage. Two, the blocks that fall land far below and litter the world spawn and creations below. These blocks remain there and never get cleaned up unless players destroy them by hand.

We've tried a couple of different work arounds now, including even placing a giant safety net (a extra layer of flooring) below the main arena that catches any falling blocks. However this is a very unideal workaround.

TL:DR

- Blocks set to destroy are simming correctly, but when landing on other floating blocks cause them to fall
- Blocks that receive blocks atop of them and are also floating are not set to simulate, aka not part of the damaged state snapshot. These blocks ignore that rule anyways and fall.
- This is problematic because floating blocks could be the floor of an arena, like in the quick start 5 minute video you have on your plugin front page. These floors should never fall or damage if they're told not to.

**Comment by kangarko** [comment]
Hello M2VC,

I am sorry for a delay in the response.

The next version of CoreArena will make the explosions more strict in regards of their location. Flying blocks will disappear if they land outside of the arena. This will also apply for falling blocks initiated by gravitational pull.

If you still experience blocks falling where they shouldn't, try expanding your arena region by one block up so the bottom ground is ignored.

Sincerely,
kangarko

**Comment by M2VC** [comment]
This issue is still happening. I'm not seeing any blocks remain outside of the arena anymore, but other blocks are still ignoring snapshot rules.

As you mentioned a work around is possible: placing the arena bounds one block up can solve this issue, but it prevents the possibility of placing mob spawners in the arena floor and so on.

**Comment by kangarko** [comment]
Please be more specific, what do you mean with "other blocks are still ignoring snapshot rules"?

**Comment by kangarko** [comment]
Closing, if you still need help please comment below.
